Orange/widgets/visualize/owscatterplot.py

```diff
@@ -59,6 +60,8 @@ class OWScatterPlot(OWWidget):
     attr_x = ContextSetting("")
     attr_y = ContextSetting("")
 
+    attr_nans = dict()
```

Comment:
This should probably be set in the constructor and re-set in `set_data` method (even if some data uses the same domain as another, it might carry a different payload).